### DISCUSSION TITLE
Running dependent job steps

### DISCUSSION THREAD

**Discussion by milosgajdos** [community/community#24954] [Actions]
<p>I have a simple <a href="https://github.com/milosgajdos/netscrape-plugins/blob/77217eecc09afdbf1c6f992263f210d07319b12f/.github/workflows/ci.yaml" rel="noopener nofollow ugc">CI workflow</a> in one of my projects that has a few steps defined in it. Recently I added a <code>docker-compose</code> to the mix as I need to run some integration tests and for that I require a DB.</p>
<p>Now, the problem I’m facing is, I need to run the testing step after the DB running in <code>docker-compose</code> has been successfully started. I noticed that sometimes a step can start before some other step that it needs hasn’t finished starting.</p>
<p>Even if I added <code>healthcheck</code> into <code>docker-compose</code> file that would still not prevent the race condition.</p>
<p>Is there a way to add some sort of step dependency between steps i.e. step2 should wait for step1 to be ready/complete.</p>
<p>Alternatively, is there some other way to do this?</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="milosgajdos" data-post="1" data-topic="164490">
<div class="title">
<div class="quote-controls"></div>
 milosgajdos:</div>
<blockquote>
<p>Is there a way to add some sort of step dependency between steps i.e. step2 should wait for step1 to be ready/complete.</p>
</blockquote>
</aside>
<p>Steps do run sequentially, what you describe sounds like the DB takes a while to be ready after the container has been started. You’ll probably have to add a ready check yourself in a step before the tests run, e.g. by probing a port or possibly checking the logs of the container for some sort of ready message.</p>

**Comment by milosgajdos** [comment]
<aside class="quote no-group" data-username="airtower-luna" data-post="2" data-topic="164490">
<div class="title">
<div class="quote-controls"></div>
 airtower-luna:</div>
<blockquote>
<p>You’ll probably have to add a ready check yourself in a step before the tests run,</p>
</blockquote>
</aside>
<p>Thanks, adding <a href="https://docs.docker.com/compose/compose-file/compose-file-v3/#healthcheck" rel="noopener nofollow ugc">healthcheck</a> did the trick, but I’ve actually moved away from compose to actions services. Both solutions work like charm.</p>
